Doc/whatsnew/3.8.rst

```diff
@@ -595,6 +595,34 @@ The :func:`ast.parse` function has some new flags:
 asyncio
 -------
 
+:func:`asyncio.run` has graduated from the provisional to stable API. This
+function can be used to execute a :term:`coroutine` and return the result while
+automatically managing the event loop. For example:
+
+    >>> import asyncio
+    >>> async def main():
+    ...     await asyncio.sleep(0)
+    ...     return 42
+    >>> asyncio.run(main())
+    42
+
+This is *roughly* equivalent to:
+
+    >>> async def main():
+    ...     await asyncio.sleep(0)
+    ...     return 42
+    >>> loop = asyncio.new_event_loop()
+    >>> asyncio.set_event_loop(loop)
+    >>> loop.run_until_complete(main())
+    42
+    >>> asyncio.set_event_loop(None)
+    >>> loop.close()
```

Comment:
change this to:

```
async def main():
    await asyncio.sleep(0)
    return 42

loop = asyncio.new_event_loop()
asyncio.set_event_loop(loop)
try:
    loop.run_until_complete()
finally:
    loop.close()
    asyncio.set_event_loop(None)
```
(no need to repl syntax in this example)

Reply:
> (no need for repl syntax in this example)

Do you want the repl syntax removed for the first one as well or just the second one?

Reply:
I'd remove it from the first one too, yes.

Reply:
In this particular case, imho, it just clutters the example. Besides, I hope people will use `python -m asyncio` when they actually need repl.

Reply:
That sounds reasonable to me, particularly with regards to not implicitly encouraging usage of the repl over `python -m asyncio`.

Reply:
@1st1 Shouldn't `loop.run_until_complete()` be `loop.run_until_complete(main())` and `loop.close()` and `asyncio.set_event_loop(None)` be the other way around?:
```
async def main():
    await asyncio.sleep(0)
    return 42

loop = asyncio.new_event_loop()
asyncio.set_event_loop(loop)
try:
    loop.run_until_complete(main())
finally:
    asyncio.set_event_loop(None)
    loop.close()
```
Only passing the coroutine function `main()` to `loop.run_until_complete()` is particularly important, I don't know that the order of `asyncio.set_event_loop(None)` and `loop.close()` actually has a functional difference (other than being closer to the real implementation).

Otherwise, I think including the `try finally` is an improvement, as it makes more similar to more similar to the real implementation. Good idea!

Reply:
> Shouldn't loop.run_until_complete() be loop.run_until_complete(main()) 

Of course, was a typo on my end.

> loop.close() and asyncio.set_event_loop(None) be the other way around?

Probably doesn't matter; let's keep it your way.

Reply:
> Of course, was a typo on my end.

Yeah I figured, I just wanted to double check that I wasn't going crazy. ;)

I had copied over the suggestion, and noticed the above points just before saving the changes.